src/main/java/com/minecolonies/coremod/entity/ai/citizen/planter/EntityAIWorkPlanter.java

```diff
@@ -217,16 +272,44 @@ private void requestPlantable(final Item current)
         }
     }
 
+    /**
+     * Check if the position is air.
+     *
+     * @param plantationItemPosition the item position data.
+     * @return true if so.
+     */
+    private boolean isItemPositionAir(final BuildingPlantation.PlantationItemPosition plantationItemPosition)
+    {
+        Block foundBlock = world.getBlockState(plantationItemPosition.getPosition().above(1)).getBlock();
+        return foundBlock instanceof AirBlock;
+    }
+
+    /**
+     * Check if the position has a valid plant.
+     *
+     * @param plantationItemPosition the item position data.
+     * @return true if so.
+     */
+    private boolean positionHasInvalidBlock(final BuildingPlantation.PlantationItemPosition plantationItemPosition)
+    {
+        Block foundBlock = world.getBlockState(plantationItemPosition.getPosition().above(1)).getBlock();
+        return plantationItemPosition.getCombination().getBlock() != foundBlock;
+    }
+
     /**
      * Check if the plant at pos it at least three high.
      *
-     * @param pos the pos to check
+     * @param plantationItemPosition the item position data.
      * @return true if so.
      */
-    private boolean isAtLeastThreeHigh(final BlockPos pos)
+    private boolean isAtLeastThreeHigh(final BuildingPlantation.PlantationItemPosition plantationItemPosition)
     {
-        return !(world.getBlockState(pos.above(1)).getBlock() instanceof AirBlock)
-                 && !(world.getBlockState(pos.above(2)).getBlock() instanceof AirBlock)
-                 && !(world.getBlockState(pos.above(3)).getBlock() instanceof AirBlock);
+        BlockPos position = plantationItemPosition.getPosition();
+        Block firstBlock = world.getBlockState(position.above(1)).getBlock();
+        if (plantationItemPosition.getCombination().getBlock() != firstBlock)
+        {
+            return false;
+        }
+        return firstBlock == world.getBlockState(position.above(2)).getBlock() && firstBlock == world.getBlockState(position.above(3)).getBlock();
```

Comment:
How about you move the height dependence into the PlantableBlockItem class thing aswell, so that one also sets the max height